### PR TITLE
Fix `toJSON()` serialisation for `co.record`

### PR DESCRIPTION
**Pull request by joeinnes**
The catchall on CoRecords was trapping the `toJSON` call, resulting in incorrect serialisation behaviour of nested CoRecords.

Tests have been added.

---

> [!NOTE]
> **Low Risk**
> Scoped to serialization/schema lookup logic and adds targeted test coverage; low risk aside from potential edge cases in proxy trapping for non-string keys.
> 
> **Overview**
> Fixes `toJSON()` serialization for `co.record` (and `co.map` containing nested records) by ensuring record item schema lookup works reliably (via `ItemsSym`) and by making `toJSON` calls on referenced CoValues resilient to proxy/catchall interception.
> 
> Adds regression tests covering `co.record.toJSON()` and `co.map` with a nested `co.record` after loading/resolution, and includes a patch changeset for `jazz-tools`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 14afa84d723ff849d4e953bda22029f3fa46b916. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/jazz-tools/src/tools/tests/coMap.test.ts [hunk 1]
```diff
@@ -414,6 +414,47 @@ describe("CoMap", async () => {
...
+      const loadedRoot = await ChatRoot.load(root.$jazz.id, {
+        resolve: {
+          chats: true,
+        },
+      });
+
+      if (!loadedRoot || !loadedRoot.$isLoaded)
+        throw new Error("Failed to load root");
+
+      console.log("Loaded root JSON:", loadedRoot.toJSON());
```
Comment (gdorsi) [inline packages/jazz-tools/src/tools/tests/coMap.test.ts]:
```suggestion
```

File: packages/jazz-tools/src/tools/tests/coMap.test.ts [hunk 2]
```diff
@@ -414,6 +414,47 @@ describe("CoMap", async () => {
...
+      });
+
+      const chat = Chat.create({ title: "General" });
+      const root = ChatRoot.create({
+        chats: {
+          general: chat,
+        },
+      });
+
+      console.log("Local root JSON:", root.toJSON());
```
Comment (gdorsi) [inline packages/jazz-tools/src/tools/tests/coMap.test.ts]:
```suggestion
```

**Review by gdorsi — APPROVED** [review]
